lua/lualine/components/branch/git_branch.lua

```diff
@@ -56,14 +56,27 @@ local function update_branch()
   branch_cache[vim.api.nvim_get_current_buf()] = current_git_branch
 end
 
+---updates the current value of current_git_branch and sets up file watch on HEAD file if value changed
+local function update_current_git_dir(git_dir)
+  if current_git_dir ~= git_dir then
+    current_git_dir = git_dir
+    update_branch()
+  end
+end
+
 ---returns full path to git directory for dir_path or current directory
 ---@param dir_path string|nil
----@return string
```

Comment:
Line 120 can return a nil value, so the linter was complaining about this annotation.